catalog-rest-service/src/main/java/org/openmetadata/catalog/Entity.java

```diff
@@ -203,6 +209,46 @@ public static void deleteEntity(String entity, UUID entityId, boolean recursive)
     dao.delete(entityId, recursive);
   }
 
+  public static <T> EntityRepository<T> getEntityRepositoryForClass(@NonNull Class<T> clazz) {
+    EntityRepository<T> entityRepository = (EntityRepository<T>) CLASS_ENTITY_REPOSITORY_MAP.get(clazz);
+    if (entityRepository == null) {
+      throw new UnhandledServerException(
+          String.format(
+              "Class %s is not an Entity class and it doesn't have a EntityRepository companion",
+              clazz.getSimpleName()));
+    }
+    return entityRepository;
+  }
+
+  /**
+   * Utility method to create the decorator from an Entity class instance. By Entity class we don't mean Entity.java but
+   * the entity classes generated from JSONSchema files like DatabaseService, Database, Table, and so on.
+   *
+   * @param entity must be an Entity class instance with a companion EntityRepository
+   * @param <T> must be the Entity class of this entity or object if multiple results are possible.
+   * @return the decorator
+   */
+  public static <T> EntityRepository<T>.EntityHandler h(@NonNull T entity) {
```

Comment:
@amiorin  can we have meaningful name here. "h" is really confusing what it does.

Reply:
Sure, do you have anything in mind? The reason for being just one letter, it's that we need to type it often and it's inspired by jQuery `$`.

We need two names for `h` (Entity->Decorator) and `r` (EntityReference->Decorator).

Reply:
To be able to use overloading we need to implement the suggestion in #2267 because of [this](https://stackoverflow.com/a/17669851).

Reply:
done